code_review.md

```diff
@@ -0,0 +1,98 @@
+
+* Is all the code easily understood? Is code unnecessarily complex ?
+* Can variable/method names be improved?
+* Are classes/files/methods becoming too large? Should they be split into smaller pieces?
+* Does it conform to the agreed coding conventions?
+* Is there any redundant or duplicate code? Is the code as modular as possible?
+* Could the design of the solution be improved (e.g. single responsibility principle, open/closed principle) ?
+* Is the code testable? (i.e. don't add too many or hide dependencies, unable to initialize objects, test frameworks can use methods, do tests exist and are they comprehensive?)
+* Does the code work? Does it perform its intended function, the logic is correct etc.
+* Things related to your area of expertiese
+
+Pro tip: Keep your own checklist of things you look for in a code review as in [9] [7].
+
+# Links
+Links from places on the Internet that helped write this:
+
+(The most useful ones are [9], [1], [3] and [2].)
+
+* [1] [Talk - Implementing a Strong Code-Review culture](https://www.youtube.com/watch?v=PJjmw9TRB7s) - the talk that inspired this.
+* [2] [Thoughtbot's Guide for Code-Review](https://github.com/thoughtbot/guides/blob/master/code-review/README.md) - straight to the point guidelines on code review. A lot of the content was taken from there.
+* [3]  [Pull request templates make code review easier](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/) - introduction to pull request templates.
+* [4] some articles on how to write good commit messages - [first](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html), [second](http://www.slideshare.net/TarinGamberini/commit-messages-goodpractices),  [third](https://wiki.openstack.org/wiki/GitCommitMessages)
+* [5] [Am I My Code?](http://mfeckie.github.io/Am-I-My-Code/) - this article explores some strategies for giving feedback when reviewing code.
+* [6] [Code review at the Dropbox iOS team](http://www.objc.io/issue-22/dropbox.html) - the article goes over their whole process of review. It has some interesting ideas.
+* [7] [Code Review Checklist](http://blog.fogcreek.com/increase-defect-detection-with-our-code-review-checklist-example/) - from Fog Creek Software
+* [8] [Effective code review tips](http://blog.fogcreek.com/effective-code-reviews-9-tips-from-a-converted-skeptic/) - from Fog Creek Software
+* [9] [Code Review Best Practices](http://kevinlondon.com/2015/05/05/code-review-best-practices.html) - really good article on code review best practices
+* [10] [11 proven practices to improve code reviews](http://www.ibm.com/developerworks/rational/library/11-proven-practices-for-peer-review/) - nice article backed by data from studies.
+* [11] [Pull request template bookmarklet](https://quickleft.com/blog/pull-request-template-bookmarklet/)
+* [12] [Pull request template chrome extension](https://github.com/sprintly/pull-request-template-chrome-extension)
```

Comment:
That's a lot of references. @justinabrahms do you think they're all as useful as the original author did?